soroban-env-host/src/test/hostile.rs

```diff
@@ -194,20 +194,132 @@ fn hostile_forged_objects_trap() -> Result<(), HostError> {
     ));
 
     // Here we call a function that passes an argument to a host function
-    // with a bad val tag. This should fail in check_val_integrity.
+    // with a bad val tag. This fails during argument unmarshalling in
+    // dispatch functions.
     let res = host.call(
         contract_id_obj,
         Symbol::try_from_small_str("badtag")?,
         host.vec_new_from_slice(&[absolute_vec.to_val()])?,
     );
     assert!(HostError::result_matches_err(
         res.clone(),
-        (ScErrorType::Value, ScErrorCode::InvalidInput)
+        (ScErrorType::WasmVm, ScErrorCode::UnexpectedType)
```

Comment:
Could this have `Value` error type instead? That would avoid the discrepancy between wasm/non-wasm contracts, and also I think is a bit more semantically correct.

Reply:
I can try. It's .. a little tricky given the error-conversion path.

Reply:
Then I guess it's not worth spending much time on this, I just thought that it could maybe be changed via a simple return value/map somewhere.

Reply:
fixed!